src/remediations/remediations.format_2.js

```diff
@@ -52,52 +37,79 @@ exports.connectionStatus = function (recipients) {
     //   - all connected direct
     //   - all disconnected direct
     //   - all no-rhc direct
-    const data = _(recipients)
-        .groupBy(recipient => {
-            switch (recipient.recipient_type) {
-                case 'satellite':
-                    return `${recipient.sat_id} ${recipient.sat_org_id}`;
 
-                default:
-                    return recipient.recipient_type;
-            }
-        })
-        .map((targets) => {
-            const recipient = targets[0];
-            switch (recipient.recipient_type) {
-                case 'satellite':
-                    return {
-                        endpoint_id: null,
-                        executor_id: recipient.sat_id,
-                        executor_type: 'RHC-satellite',
-                        executor_name: `Satellite ${recipient.sat_id} Org ${recipient.sat_org_id}`,
-                        system_count: recipient.systems.length,
-                        connection_status: recipientStatusMap[recipient.status]
-                    };
-
-                case 'directConnect':
-                    return {
-                        endpoint_id: null,
-                        executor_id: null,
-                        executor_type: 'RHC',
-                        executor_name: null,
-                        system_count: targets.length,
-                        connection_status: computeStatus(targets)
-                    };
-
-                default:
-                    return {
-                        endpoint_id: null,
-                        executor_id: null,
-                        executor_type: 'None',
-                        executor_name: null,
-                        system_count: recipient.systems.length,
-                        connection_status: recipientStatusMap[recipient.status]
-                    };
-            }
-        })
-        .sortBy('executor_name')
-        .value();
+    const data = _(recipients)
+    .groupBy(recipient => {
+        // key will either be one of recipientStatusMap or a recipient_id UUID
```

Comment:
There is a combination of two unrelated things when doing the grouping. I'd suggest to filter or group by type, and then operate on each of the type separately.

Reply:
I'll restructure this a bit.  It seems to be slightly faster to handle these two cases together, as part of one stream, than to split them out and then rejoin them.  I also find the code easier to follow this way but that's just a personal preference :-)